### ISSUE TITLE
Lab 3 peer review - Parameters tuning

### ISSUE THREAD

**Issue by simocosimo** [SamuelePino/computational-intelligence#3]
Hi :)

## Main comment
I'm going to review the RL algorithm, it seems that you focused more on this one.
Very nice use of the state-move pair to index the g table, it is useful in order to make the model a little less "overfitted" on the learning phase opponent (I used just the state and in fact noticed that things changed quite a bit when the starting player changed, for example).

It's weird that you get such different results between runs, I cannot really spot the error. Maybe it could help to better manage the random factor (I also see that you pick a random move when you haven't got a state in the g table) or try increasing the number of episodes in order to have a better learning phase.

But again, this is just a matter of tuning parameters in order to get better results. For example, I noticed that a high learning rate worked better than a small one with Nim.

Nice job, though 👍🏼
Good luck with the exam!
